.gitlab-ci-check-golang-dockerimage.yml

```diff
@@ -0,0 +1,64 @@
+# .gitlab-ci-check-golang-dockerimage.yml
+#
+# This gitlab-ci template builds a Docker image assuming a single Dockerfile
+# and publish it on Docker Hub
+#
+# Add it to the project in hand through Gitlab's include functionality
+#
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-check-golang-dockerimage.yml'
+#
+# Requires the following variables set in the project CI/CD settings:
+# DOCKER_HUB_USERNAME: Username for docker.io
+# DOCKER_HUB_PASSWORD: Password for docker.io
+#
+
+stages:
+  - build
+  - publish
+
+build:
+  stage: build
+  tags:
+    - docker
+  image: docker
+  services:
+    - docker:dind
+  before_script:
+    - export DOCKER_REPOSITORY="mendersoftware/${CI_PROJECT_NAME}"
```

Comment:
very nice! didn't think of it at the time and assumed the project names would diverge